### PR TITLE
Convert project to babel

### PR DESCRIPTION
**Pull request by timmywil**
- Includes code coverage with isparta
- Pre-compiles for production (main: dist/Injector.js)
- npm scripts added for development (npm run build, npm run dev (watching), npm run lint, npm test, npm run coveralls).
- Still includes coffee-script if needed for injecting (note that we are still testing injecting CS modules)
- package.json now specifies a `files` property for whitelisting what should be included on npm install (mainly, the dist folder). Note that some files are always included (like package.json, README.md, LICENSE) regardless of whether they are in the files array.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -20,8 +17,5 @@ insert_final_newline = false
 [Makefile]
 indent_style = tab
 
-[*.{js,json,coffee,less,sass,scss,css,rb}]
-indent_size = 2
```
Comment (acolchado) [inline .editorconfig:24]: Why is this removed?

Comment (timmywil) [inline .editorconfig:24]: Redundancy. The default indent size is 2.

Comment (morewry) [inline .editorconfig:24]: I thought the reason for having it is that I might have set my default indent size to 4 in my editor, and this ensures that for this codebase it gets set to 2. @morewry

File: src/CallChain.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import chalk from "chalk"
+
+class CallChain {
+  constructor(name1, parent) {
```
Comment (morewry) [inline src/CallChain.js]: Why `name1`?

File: src/CallChain.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  add(name) {
+    return new CallChain(name, this)
+  }
+
+  hasParentName(name) {
+    return this.name === name || (this.parent && this.parent.hasParentName(name))
+  }
+
+  hasCyclic() {
+    const parent = this.parent
```
Comment (morewry) [inline src/CallChain.js:22]: What does this do in this method? Does it do something different in terms of performance or anything than using `this.parent` on lines 23 and 24?

File: src/CallChain.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  getHighlightedName() {
+    return chalk.yellow(this.name)
+  }
+
+  getPath(highlight) {
+    let path = ""
+    if (this.parent) {
+      path = `${this.parent.getPath()} -> `
+    }
+    return (path += (highlight ? this.getHighlightedName() : this.name))
```
Comment (morewry) [inline src/CallChain.js]: Personally, I find this more confusing to read than the original more expanded form.

File: src/ContainerManagement.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
+import Dependency from "./Dependency"
+
+const rall = /.+/
+
+export default {
```
Comment (morewry) [inline src/ContainerManagement.js:5]: Do we no longer need to do:

``` javascript
this.dependencies = {}
```

in order to do 

``` javascript
this.dependencies[name] = "hello world";
```

Shouldn't that `Cannot set property __ of undefined`?

File: .editorconfig (hunk 1 above)
Comment (timmywil) [inline .editorconfig:24]: @morewry Yup, it will still work that way, as long as you have an editorconfig plugin. It's not that it's redundant with my or someone else's editor. It's that this setting is already set [a few lines above](https://github.com/timmywil/spur-ioc/blob/babel/.editorconfig#L5).

File: src/CallChain.js (hunk 3 above)
Comment (timmywil) [inline src/CallChain.js:22]: Not really. Property access can be expensive depending on the property, so I've adopted the practice, but I doubt it makes a difference in this case.

File: src/CallChain.js (hunk 2 above)
Comment (timmywil) [inline src/CallChain.js]: Inserted by lebab. I'll change.

File: src/CallChain.js (hunk 4 above)
Comment (timmywil) [inline src/CallChain.js]: Fair enough.

File: src/ContainerManagement.js (hunk 5 above)
Comment (timmywil) [inline src/ContainerManagement.js:5]: I'm treating this file more like a mixin/interface than it's own class (since JS doesn't have multiple inheritance). We were faking it by looping over everything and calling the different constructors, but ES6 prohibits that. `this.dependencies` is instead initialized in the single constructor in [Injector.js](https://github.com/timmywil/spur-ioc/blob/babel/src/Injector.js#L14). Had I written this in ES6 originally, I probably would have encapsulated purposes and done things like `this.containerManager.method`, rather than inheriting, but I wasn't sure what that would break and this was just the first step.

File: .editorconfig (hunk 1 above)
Comment (morewry) [inline .editorconfig:24]: Oh, I see! Thanks :-)

**Comment by timmywil** [comment]
:clap:

**Comment by acolchado** [comment]
:)
